test/kldivergences.jl

```diff
@@ -0,0 +1,86 @@
+@testset "KL divergences" begin
+    function test_kl(p, q)
+        @test kldivergence(p, q) > 0
+        @test kldivergence(p, p) ≈ 0 atol=1e-1
+        @test kldivergence(q, q) ≈ 0 atol=1e-1
+        if p isa UnivariateDistribution
+            @test kldivergence(p, q) ≈ invoke(kldivergence, Tuple{UnivariateDistribution,UnivariateDistribution}, p, q) atol=1e-1
+        elseif p isa MultivariateDistribution
+            @test kldivergence(p, q) ≈ invoke(kldivergence, Tuple{MultivariateDistribution,MultivariateDistribution}, p, q; nsamples=10000) atol=1e-1
+        end
+    end
+    
+    @testset "Categorical" begin
+        @test kldivergence(Categorical([0.0, 0.1, 0.9]), Categorical([0.1, 0.1, 0.8])) ≥ 0
+        @test kldivergence(Categorical([0.0, 0.1, 0.9]), Categorical([0.1, 0.1, 0.8])) ≈
+            kldivergence([0.0, 0.1, 0.9], [0.1, 0.1, 0.8])
+        @test expectation(Normal(0.0, 1.0), identity, 1e-10) ≤ 1e-9
+    end
+
+    @testset "univariate" begin
+        @testset "Beta" begin
+            p = Beta(2, 10)
+            q = Beta(3, 5)
+            test_kl(p, q)
+        end
+        @testset "Exponential" begin
+            p = Exponential(2.0)
+            q = Exponential(3.0)
+            test_kl(p, q)
+        end
+        @testset "Gamma" begin
+            p = Gamma(2.0, 1.0)
+            q = Gamma(3.0, 2.0)
+            test_kl(p, q)
+        end
+        @testset "InverseGamma" begin
+            p = InverseGamma(2.0, 1.0)
+            q = InverseGamma(3.0, 2.0)
+            test_kl(p, q)
+        end
+        @testset "Normal" begin
+            p = Normal(0, 1)
+            q = Normal(0.5, 0.5)
+            test_kl(p, q)
+        end
+        @testset "Poisson" begin
+            p = Poisson(4.0)
+            q = Poisson(3.0)
+            test_kl(p, q)
+
+            p_0 = Poisson(0.0)
+            test_kl(p_0, p)
+        end
+    end
+    @testset "multivariate" begin
+        # We try on a newly created AbstractMvNormal
+        struct CholeskyMvNormal{TL,Tm} <: Distributions.AbstractMvNormal
+            m::Tm
+            L::TL
+        end
```

Comment:
Julia 1.0 doesn't allow type definitions in a local scope. Can you move this outside of the testsets?